G-Research.OpenTelemetry.Processor.Partial/PartialActivityProcessor.cs

```diff
@@ -17,29 +16,42 @@ public class PartialActivityProcessor : BaseProcessor<Activity>
     private readonly int _initialHeartbeatDelayMilliseconds;
     private readonly int _processIntervalMilliseconds;
 
-    private readonly ConcurrentDictionary<ActivitySpanId, Activity> _activeActivities;
+    private readonly object _lock = new();
 
-    private readonly ConcurrentQueue<(ActivitySpanId SpanId, DateTime InitialHeartbeatTime)>
+    private readonly Dictionary<ActivitySpanId, Activity> _activeActivities;
+
+    private readonly Queue<(ActivitySpanId SpanId, DateTime InitialHeartbeatTime)>
         _delayedHeartbeatActivities;
 
-    private readonly ConcurrentDictionary<ActivitySpanId, bool> _delayedHeartbeatActivitiesLookup;
+    private readonly HashSet<ActivitySpanId> _delayedHeartbeatActivitiesLookup;
 
-    private readonly ConcurrentQueue<(ActivitySpanId SpanId, DateTime NextHeartbeatTime)>
+    private readonly Queue<(ActivitySpanId SpanId, DateTime NextHeartbeatTime)>
         _readyHeartbeatActivities;
 
     // added for tests convenience
-    public IReadOnlyDictionary<ActivitySpanId, Activity> ActiveActivities => _activeActivities;
+    public IReadOnlyDictionary<ActivitySpanId, Activity> ActiveActivities()
+    {
+        return new Dictionary<ActivitySpanId, Activity>(_activeActivities);
+    }
```

Comment:
The copies of these colelctions still need to be taken inside the lock (for all these properties).

```suggestion
    public IReadOnlyDictionary<ActivitySpanId, Activity> ActiveActivities()
    {
        lock(_lock)
        {
            return new Dictionary<ActivitySpanId, Activity>(_activeActivities);
        }
    }
```